src/main/java/cherry/main/Deadline.java

```diff
@@ -0,0 +1,90 @@
+package cherry.main;
+
+import cherry.utils.Day;
+
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+
+/**
+ * Represents a task with a specific deadline.
+ * The deadline can be either a specific date (in the format yyyy-MM-dd) or a day of the week.
+ */
+public class Deadline extends Task {
+
+    protected Object by;
```

Comment:
```suggestion
    protected Object byDeadline;
```
I think that changing the variable name here could make it clearer for understanding